streams/src/main/java/com/palantir/common/streams/MoreCollectors.java

```diff
@@ -61,6 +61,18 @@ public final class MoreCollectors {
                 ImmutableList.Builder::build);
     }
 
+    /**
+     * Collect a Stream of Map.Entry (e.g. a StreamEx EntryStream) into a Guava ImmutableMap, which preserves iteration
+     * order. Duplicate keys will result in an error.
+     *
+     * This is just a convenience method for Guava's built-in {@link ImmutableMap#toImmutableMap}.
+     *
+     * Beware that {@code EntryStream#toImmutableMap()} does NOT preserve iteration order, as it uses a regular HashMap.
+     */
+    public static <K, V> Collector<Map.Entry<K, V>, ?, ImmutableMap<K, V>> toImmutableMap() {
+        return ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue);
```

Comment:
It might be worth noting that because this is just syntactic sugar around `ImmutableMap#toImmutableMap` that this will throw if keys and/or values are `null`

Reply:
updated!